espnet2/asr/espnet_model.py

```diff
@@ -209,9 +211,26 @@ def forward(
             for layer_idx, intermediate_out in intermediate_outs:
                 # we assume intermediate_out has the same length & padding
                 # as those of encoder_out
-                loss_ic, cer_ic = self._calc_ctc_loss(
-                    intermediate_out, encoder_out_lens, text, text_lengths
-                )
+
+                # use auxillary ctc data if specified
+                loss_ic = None
+                if self.aux_ctc is not None:
+                    idx_key = str(layer_idx)
+                    if idx_key in self.aux_ctc:
+                        aux_data_key = self.aux_ctc[idx_key]
+                        aux_data_tensor = kwargs.get(aux_data_key, None)
+                        aux_data_lengths = kwargs.get(aux_data_key+"_lengths", None)
+                    
+                        if aux_data_tensor is not None and aux_data_lengths is not None:
```

Comment:
Should it throw an exception if `aux_data_tensor` is `None` but `self.aux_ctc` is not `None`. This would mean an auxiliary task was defined but no data was found. The current implementation defaults it to use the ASR label instead, so if you forget to add the auxiliary data the program wont crash.